sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1265,12 +1266,15 @@ mod test_cluster_dns_ip {
     }
 }
 
+type EnvVarMap = HashMap<SingleLineString, SingleLineString>;
+
 /// CredentialProvider contains the settings for a credential provider for use
 /// in CredentialProviderConfig.
-#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub struct CredentialProvider {
     enabled: bool,
     image_patterns: Vec<SingleLineString>,
     cache_duration: Option<KubernetesDurationValue>,
+    env_variables: Option<EnvVarMap>,
```

Comment:
Probably doesn't matter here, but these will be randomly ordered when iterated for the config file.

Reply:
Yep, it shouldn't matter what order these come in as in the final rendering of the template. They all just end up being environment variables in the credential process.

I briefly considered using a `BTreeMap`, but it ultimately doesn't matter.